core/templates/dev/head/domain/collection/SkillListObjectFactory.js

```diff
@@ -111,11 +108,11 @@ oppia.factory('SkillListObjectFactory', [function() {
     return true;
   };
 
-  // Removes a skill based on a search name. Returns whether or not this was
-  // successful, which depends on whether the given skill name is actually in
+  // Removes a skill based on an ID. Returns whether or not this was
+  // successful, which depends on whether the given skill ID is actually in
   // the skills list.
-  SkillList.prototype.removeSkillByName = function(skillName) {
-    var index = this.indexOfSkill(skillName);
+  SkillList.prototype.removeSkillById = function(skillId) {
+    var index = this.indexOfSkill(skillId);
```

Comment:
I'm not refactoring the SkillList-related naming for now, since it might get removed entirely.